In [5]:
import pandas as pd
from datatable import dt, f

TODO: Deixar aqui apenas os dados brutos

# Aquisição dos dados brutos

Primeiramente, conforme falado na seção anterior, unimos os microdados de todos os anos para ter um olhar menos fechado em um único ano.

In [9]:
def get_path_table_by_year(year):
  path_table = '/'.join([
    '../data/raw',
    'censo_{year}.zip',
    'Microdados do Censo da Educaç╞o Superior {year}',
    'dados',
    'MICRODADOS_CADASTRO_CURSOS_{year}.CSV'
  ]).format(year = year)
  return path_table

In [29]:
def aggregate_raw_data(from_year, to_year):
  for y in range(from_year, to_year + 1):
    path_table = get_path_table_by_year(y)
    dataset = dt.fread(path_table)
  return dataset

In [30]:
df_raw = aggregate_raw_data(2010, 2019)
display(df_raw)

<Frame#7fbdd3e7d570 246081x196>

Para salvar esses dados agregados em um csv, é necessário fazer o seguinte:

In [31]:
df_raw.to_csv('../data/raw/aggregated_dataset.csv')

In [101]:
# Transformando datatable em dataframe do pandas para facilitar
df = df_raw.to_pandas()
df.shape

(246081, 196)

Após a agregação, a tabela bruta para trabalho ficou com 246,081 linhas e 196 colunas.  
Após isso, era necessário entender mais sobre as principais características e atributos da base que nos seriam interessantes.  
Para isso, é necessário conhecer a base primeiro, o que será feito nas células seguintes:

In [103]:
# Permitindo todas as colunas
pd.set_option("display.max.columns", None)
# Conhecendo o dataset a partir de alguns poucos registros
df.head(30)

NU_ANO_CENSO     NO_REGIAO  CO_REGIAO        NO_UF SG_UF  CO_UF  \
0           2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
1           2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
2           2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
3           2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
4           2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
5           2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
6           2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
7           2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
8           2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
9           2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
10          2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
11          2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
12          2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
13          2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
14          2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
15          2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
16          2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
17          2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
18          2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
19          2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
20          2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
21          2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
22          2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
23          2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
24          2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
25          2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
26          2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
27          2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
28          2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   
29          2019  Centro-Oeste        5.0  Mato Grosso    MT   51.0   

       NO_MUNICIPIO  CO_MUNICIPIO IN_CAPITAL  TP_ORGANIZACAO_ACADEMICA  \
0          Água Boa      5.10e+06      False                         1   
1          Água Boa      5.10e+06      False                         1   
2          Água Boa      5.10e+06      False                         1   
3     Alto Araguaia      5.10e+06      False                         1   
4          Aripuanã      5.10e+06      False                         1   
5          Aripuanã      5.10e+06      False                         1   
6          Aripuanã      5.10e+06      False                         1   
7   Barra do Garças      5.10e+06      False                         1   
8   Barra do Garças      5.10e+06      False                         1   
9   Barra do Garças      5.10e+06      False                         1   
10  Barra do Garças      5.10e+06      False                         1   
11  Barra do Garças      5.10e+06      False                         1   
12  Barra do Garças      5.10e+06      False                         1   
13  Barra do Garças      5.10e+06      False                         1   
14  Barra do Garças      5.10e+06      False                         1   
15         Canarana      5.10e+06      False                         1   
16         Canarana      5.10e+06      False                         1   
17          Colíder      5.10e+06      False                         1   
18           Cuiabá      5.10e+06       True                         1   
19           Cuiabá      5.10e+06       True                         1   
20           Cuiabá      5.10e+06       True                         1   
21           Cuiabá      5.10e+06       True                         1   
22           Cuiabá      5.10e+06       True                         1   
23           Cuiabá      5

In [106]:
# Conhecendo a base
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 246081 entries, 0 to 246080
Columns: 196 entries, NU_ANO_CENSO to QT_CONC_MOB_ACADEMICA
dtypes: bool(5), float64(4), int32(177), object(10)
memory usage: 193.6+ MB


In [127]:
# Descobrindo as colunas
df.columns.values

array(['NU_ANO_CENSO', 'NO_REGIAO', 'CO_REGIAO', 'NO_UF', 'SG_UF',
       'CO_UF', 'NO_MUNICIPIO', 'CO_MUNICIPIO', 'IN_CAPITAL',
       'TP_ORGANIZACAO_ACADEMICA', 'TP_CATEGORIA_ADMINISTRATIVA',
       'TP_REDE', 'CO_IES', 'NO_CINE_ROTULO', 'CO_CINE_ROTULO',
       'CO_CINE_AREA_GERAL', 'NO_CINE_AREA_GERAL',
       'CO_CINE_AREA_ESPECIFICA', 'NO_CINE_AREA_ESPECIFICA',
       'CO_CINE_AREA_DETALHADA', 'NO_CINE_AREA_DETALHADA',
       'TP_GRAU_ACADEMICO', 'TP_MODALIDADE_ENSINO', 'TP_NIVEL_ACADEMICO',
       'QT_CURSO', 'QT_VG_TOTAL', 'QT_VG_TOTAL_DIURNO',
       'QT_VG_TOTAL_NOTURNO', 'QT_VG_TOTAL_EAD', 'QT_VG_NOVA',
       'QT_VG_PROC_SELETIVO', 'QT_VG_REMANESC', 'QT_VG_PROG_ESPECIAL',
       'QT_INSCRITO_TOTAL', 'QT_INSCRITO_TOTAL_DIURNO',
       'QT_INSCRITO_TOTAL_NOTURNO', 'QT_INSCRITO_TOTAL_EAD',
       'QT_INSC_VG_NOVA', 'QT_INSC_PROC_SELETIVO', 'QT_INSC_VG_REMANESC',
       'QT_INSC_VG_PROG_ESPECIAL', 'QT_ING', 'QT_ING_FEM', 'QT_ING_MASC',
       'QT_ING_DIURNO', 'QT_ING_NOTURNO',

In [84]:
# Descobrindo as Áreas de Conhecimento Específico
specific_areas = df.drop_duplicates(subset=['CO_CINE_AREA_ESPECIFICA','NO_CINE_AREA_ESPECIFICA'], ignore_index=True)
print("Área Específica de Conhecimento")
specific_areas[['CO_CINE_AREA_ESPECIFICA','NO_CINE_AREA_ESPECIFICA']]

Área Específica de Conhecimento


CO_CINE_AREA_ESPECIFICA                            NO_CINE_AREA_ESPECIFICA
0                        11                                           Educação
1                        41                           Negócios e administração
2                        32                           Comunicação e informação
3                        42                                            Direito
4                        61  Computação e Tecnologias da Informação e Comun...
5                        73                           Arquitetura e construção
6                        81                                        Agricultura
7                        91                                              Saúde
8                        21                                              Artes
9                        22                       Humanidades (exceto línguas)
10                       31                 Ciências sociais e comportamentais
11                       51                   Ciências biológicas e correlatas
12                       53                                   Ciências físicas
13                       54                           Matemática e estatística
14                       71                 Engenharia e profissões correlatas
15                       72                           Produção e processamento
16                       82                                       Silvicultura
17                       84                                        Veterinária
18                       92                                          Bem-estar
19                        1                                  Programas básicos
20                       23                                            Línguas
21                       52                                      Meio ambiente
22                      101                                  Serviços pessoais
23                       83                                              Pesca
24                       78  Programas interdisciplinares abrangendo engenh...
25                       28  Programas interdisciplinares abrangendo artes ...
26                       68  Programas interdisciplinares abrangendo comput...
27                      103                              Serviços de segurança
28                      104                             Serviços de transporte
29                      102            Higiene e serviços de saúde ocupacional
30                       58  Programas interdisciplinares abrangendo ciênci...
31                       98  Programas interdisciplinares abrangendo saúde ...
32                       38  Programas interdisciplinares abrangendo ciênci...
33                       18   Programas interdisciplinares abrangendo educação
34                       88  Programas interdisciplinares abrangendo agricu...
35                       48  Programas interdisciplinares abrangendo negóci...

In [80]:
# Descobrindo as Áreas de Conhecimento Geral
areas = df.drop_duplicates(subset=['CO_CINE_AREA_GERAL','NO_CINE_AREA_GERAL'], ignore_index=True)
print("Área Geral de Conhecimento")
areas[['CO_CINE_AREA_GERAL','NO_CINE_AREA_GERAL']]

Área Geral de Conhecimento


CO_CINE_AREA_GERAL                                 NO_CINE_AREA_GERAL
0                    1                                           Educação
1                    4                  Negócios, administração e direito
2                    3         Ciências sociais, comunicação e informação
3                    6  Computação e Tecnologias da Informação e Comun...
4                    7                  Engenharia, produção e construção
5                    8     Agricultura, silvicultura, pesca e veterinária
6                    9                                  Saúde e bem-estar
7                    2                                Artes e humanidades
8                    5        Ciências naturais, matemática e estatística
9                    0                                  Programas básicos
10                  10                                           Serviços